AutomatticTracks/src/sharedTest/kotlin/com/automattic/android/tracks/fakes/FakeDataProvider.kt

```diff
@@ -13,17 +15,21 @@ class FakeDataProvider(
     override val releaseName: String = "testReleaseName",
     override val locale: Locale? = Locale.US,
     override val enableCrashLoggingLogs: Boolean = true,
-    var user: CrashLoggingUser? = testUser1,
     var crashLoggingEnabled: Boolean = true,
     var shouldDropException: (String, String, String) -> Boolean = { _: String, _: String, _: String -> false },
     var extraKeys: List<String> = emptyList(),
     var provideExtrasForEvent: (Map<ExtraKnownKey, String>) -> Map<ExtraKnownKey, String> = { currentExtras -> currentExtras },
-    var applicationContext: Map<String, String> = emptyMap(),
+    initialUser: CrashLoggingUser? = testUser1,
+    initialApplicationContext: Map<String, String> = emptyMap(),
 ) : CrashLoggingDataProvider {
 
-    override fun userProvider(): CrashLoggingUser? {
-        return user
-    }
+    val fakeUserEmitter = MutableStateFlow(initialUser)
+
+    val fakeApplicationContextEmitter = MutableStateFlow(initialApplicationContext)
+
+    override val user: Flow<CrashLoggingUser?> = fakeUserEmitter
+
+    override val applicationContextProvider: Flow<Map<String, String>> = fakeApplicationContextEmitter
```

Comment:
You can see a similar approach in `FakeDataSource` in [Android Testing docs](https://developer.android.com/kotlin/flow/test#continuous-collection)